zabbix_metrics_watcher.py

```diff
@@ -214,14 +216,32 @@ def run(zabbix_host, osbs_master, config, instance):
 
             logger.info("Found build %s in %s", build_name, status)
             build = Build(build_name, cmd_base)
-            if status == 'Pending':
+            if status == 'New':
+                if list(builds_in_new) == []:
+                    builds_in_new_time = datetime.datetime.now()
+                else:
+                    timediff = int((datetime.datetime.now() - builds_in_new_time).total_seconds())
+                    logger.info("%s build(s) are in New for %s sec", len(builds_in_new), timediff)
+                    _send_zabbix_message(zabbix_host, osbs_master, "new_duration", timediff)
```

Comment:
Thats sounds fair, although I'm not sure what info should be stored in Zabbix then - a build name and duration in New? I don't think its worth it, as usually either all builds pass or all stuck in New - in this case a total time spent in New is sufficient.

A threshold to avoid a zabbix message is a good idea - I think 5 secs for a start would do